app/src/main/java/org/mozilla/tv/firefox/ScreenController.kt

```diff
@@ -53,53 +79,114 @@ class ScreenController {
     }
 
     fun showSettingsScreen(fragmentManager: FragmentManager) {
-        val settingsFragment = SettingsFragment.create()
-        fragmentManager.beginTransaction()
-                .replace(R.id.container, settingsFragment, SettingsFragment.FRAGMENT_TAG)
-                .addToBackStack(null)
-                .commit()
+        handleTransition(fragmentManager, ScreenControllerStateMachine.Transition.ADD_SETTINGS)
+        stateMachine.settingsOpened()
+    }
+
+    fun showPocketScreen(fragmentManager: FragmentManager) {
+        handleTransition(fragmentManager, ScreenControllerStateMachine.Transition.ADD_POCKET)
+        stateMachine.pocketOpened()
     }
 
     fun showBrowserScreenForCurrentSession(fragmentManager: FragmentManager, session: Session) {
-        val fragment = fragmentManager.findFragmentByTag(WebRenderFragment.FRAGMENT_TAG) as WebRenderFragment?
-        if (fragment != null && fragment.session == session) {
-            // There's already a BrowserFragment displaying this session.
-            return
+        if (session.url != URLs.APP_URL_HOME) {
```

Comment:
I wonder if we should centralize how we handle APP_URL_HOME: currently, the constant is sprinkled throughout the code and there are two major places it's handled:
- ScreenController.showBrowserScreenForCurrentSession (called only from MainActivity.onCreate): this exposes the web render fragment (and actually hides the nav overlay)
- WebRenderFragment.onUrlChanged: this adds the navigation overlay on the screen

^ Tracking down how the home screen is displayed is more confusing than I'd like. Maybe `isUrlHome` should become a state in the state machine or we can find a way to centralize the code. This can be a follow-up (though if it's not going to get done soon, maybe it's not worth filing at all like other aspirational refactor bugs).

Note: not enough brain to really understand the implications of this code but I'll let you think about it so you can get the review done. 😝

Reply:
I made this a private function, so we could consider making it publicly accessible.

Thinking about who would need to know about `isOnHomeUrl` - it doesn't look like a lot of other places actually check the session url against APP_URL_HOME, but rather check input against it.

Reply:
The redundant show/hide overlay on about:home should also be covered in #1644